### ISSUE TITLE
Provide a way to migrate code that still allows using git diff tooling to see what was changed

### ISSUE THREAD

**Issue by danroth27** [dotnet/systemweb-adapters#224]
One of the downsides of upgrading an app by migrating code to a new project is that you lose the ability to see what code changes were made by the tooling using source code diff tooling like `git diff`. Could we have a way for the developer to create a commit before the code changes are applied so that they can diff the before and after state of the code?

@sayedihashimi @antonfirsov

**Comment by abpiskunov** [comment]
interesting idea, need to think about the flow.. So far it sounds pretty fragile: git commit should happen right after files copied in the middle of migration, what if user had some other pending changes? Commit should happen automatically I assume. 
Instead of git commit, could we just have a diff window pointing to new and old files? e could reuse some roslyn diffing infrastructure I hope to point it to before and after files. It could be a gesture in the Migration results window for each file or somewhere else... works without git too ;).  Thoughts?

**Comment by danroth27** [comment]
I don't think we want to create a git commit for the developer. But maybe we should have a way to move over the files and apply code transformations using two separate gestures so the user can add a source control commit in between?
